src/plugins/visualizations/public/expressions/visualization_renderer.tsx

```diff
@@ -57,6 +57,10 @@ export const visualization = (): ExpressionRenderDefinition<VisRenderValue> => (
       unmountComponentAtNode(domNode);
     });
 
+    if (vis.getUiState() !== uiState) {
+      vis.setUiState(uiState);
```

Comment:
I am concerned that `handlers.uiState` has a type of `any` while `vis.setUiState` is expecting `PersistedState`.

Reply:
You're right, but I think it's not a big issue. According the implementation in the [Visualization component](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/visualizations/public/components/visualization.tsx#L63), the `uiState` has already been assigned to the vis object. The `vis.uiState` always be an empty since `vis` object recreated in every render. To address this `any` concern, how about add a type assertment in the compare condition. Refactor code like below:
```ts
if (uiState instanceof PersistedState && vis.getUiState() !== uiState) {
      vis.setUiState(uiState);
}
```

Reply:
Nit: Should we do a deep compare?

Reply:
`vis.getUiState()` should always be an empty object in  this place. Since we construct a new `ExpVis` object here. The `uiState` of this object should be empty too.